### PR TITLE
OAK-9741 Test cases for invalid index definitions

### PR DESCRIPTION
**Pull request by thomasmueller**

### CODE REVIEW

**Comment by nit0906** [comment]
There seems to be compilation errors in the ci builds - 

````
[](https://ci-builds.apache.org/blue/organizations/jenkins/Jackrabbit%2Foak-trunk-pr/detail/PR-532/1/pipeline#step-422-log-1735)[](https://ci-builds.apache.org/blue/organizations/jenkins/Jackrabbit%2Foak-trunk-pr/detail/PR-532/1/pipeline#step-422-log-1736)[](https://ci-builds.apache.org/blue/organizations/jenkins/Jackrabbit%2Foak-trunk-pr/detail/PR-532/1/pipeline#step-422-log-1737)[](https://ci-builds.apache.org/blue/organizations/jenkins/Jackrabbit%2Foak-trunk-pr/detail/PR-532/1/pipeline#step-422-log-1738)[](https://ci-builds.apache.org/blue/organizations/jenkins/Jackrabbit%2Foak-trunk-pr/detail/PR-532/1/pipeline#step-422-log-1739)[](https://ci-builds.apache.org/blue/organizations/jenkins/Jackrabbit%2Foak-trunk-pr/detail/PR-532/1/pipeline#step-422-log-1740)[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:testCompile (default-testCompile) on project oak-lucene: Compilation failure: Compilation failure: 

[ERROR] /home/jenkins/jenkins-agent/workspace/Jackrabbit_oak-trunk-pr_PR-532/oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/invalidData/InvalidIndexDefinitionTest.java:[166,61] cannot find symbol

[ERROR]   symbol:   method of(java.lang.String)

[ERROR]   location: interface java.util.List

[ERROR] /home/jenkins/jenkins-agent/workspace/Jackrabbit_oak-trunk-pr_PR-532/oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/invalidData/InvalidIndexDefinitionTest.java:[167,61] cannot find symbol

[ERROR]   symbol:   method of(java.lang.String)

[ERROR]   location: interface java.util.List
````

But I see that java.util.List is imported - not sure why it fails then though.

File: oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/invalidData/InvalidIndexDefinitionTest.java [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+        Tree def = createIndexNodeAndData();
+        // 1L + Integer.MAX_VALUE results in IllegalArgumentException: Out of range: 2147483648
+        def.setProperty("blobSize", 1L + Integer.MAX_VALUE);
+        root.commit();
+    }    
+
+    @Test
+    public void negativeBlobSize() throws CommitFailedException {
+        Tree def = createIndexNodeAndData();
+        // 1L + Integer.MAX_VALUE results in IllegalArgumentException: Out of range: 2147483648
```
Comment (nit0906) [inline oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/invalidData/InvalidIndexDefinitionTest.java]: Comment can be fixed here to point out the expected behaviour when this value is set to a negative number.

File: oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/invalidData/InvalidIndexDefinitionTest.java [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+        def.setProperty(LuceneIndexConstants.EVALUATE_PATH_RESTRICTION, "abc");
+        root.commit();
+        String query = "select [jcr:path] from [nt:base] where isdescendantnode('/tmp') and upper([test]) = 'HELLO'";
+        assertThat(explain(query), containsString("lucene:test"));
+        assertQuery(query, Lists.newArrayList("/tmp/testNode"));
+    }
+
+    @Test
+    public void invalidIncludedPath() throws CommitFailedException {
+        // this will commit, but in oak-run it will not work:
```
Comment (nit0906) [inline oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/invalidData/InvalidIndexDefinitionTest.java:165]: Maybe we can add assertions on logs here to check this error is logged properly ? 
Refer LogCustomizer usage here https://github.com/apache/jackrabbit-oak/blob/trunk/oak-core/src/test/java/org/apache/jackrabbit/oak/query/stats/QuerySimilarCostTest.java

**Review by nit0906 — CHANGES_REQUESTED** [review]
lgtm, just a couple of minor things pointed out.

**Comment by thomasmueller** [comment]
> There seems to be compilation errors in the ci builds -
> 
> ```
> [](https://ci-builds.apache.org/blue/organizations/jenkins/Jackrabbit%2Foak-trunk-pr/detail/PR-532/1/pipeline#step-422-log-1735)[](https://ci-builds.apache.org/blue/organizations/jenkins/Jackrabbit%2Foak-trunk-pr/detail/PR-532/1/pipeline#step-422-log-1736)[](https://ci-builds.apache.org/blue/organizations/jenkins/Jackrabbit%2Foak-trunk-pr/detail/PR-532/1/pipeline#step-422-log-1737)[](https://ci-builds.apache.org/blue/organizations/jenkins/Jackrabbit%2Foak-trunk-pr/detail/PR-532/1/pipeline#step-422-log-1738)[](https://ci-builds.apache.org/blue/organizations/jenkins/Jackrabbit%2Foak-trunk-pr/detail/PR-532/1/pipeline#step-422-log-1739)[](https://ci-builds.apache.org/blue/organizations/jenkins/Jackrabbit%2Foak-trunk-pr/detail/PR-532/1/pipeline#step-422-log-1740)[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:testCompile (default-testCompile) on project oak-lucene: Compilation failure: Compilation failure: 
> 
> [ERROR] /home/jenkins/jenkins-agent/workspace/Jackrabbit_oak-trunk-pr_PR-532/oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/invalidData/InvalidIndexDefinitionTest.java:[166,61] cannot find symbol
> 
> [ERROR]   symbol:   method of(java.lang.String)
> 
> [ERROR]   location: interface java.util.List
> 
> [ERROR] /home/jenkins/jenkins-agent/workspace/Jackrabbit_oak-trunk-pr_PR-532/oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/invalidData/InvalidIndexDefinitionTest.java:[167,61] cannot find symbol
> 
> [ERROR]   symbol:   method of(java.lang.String)
> 
> [ERROR]   location: interface java.util.List
> ```
> 
> But I see that java.util.List is imported - not sure why it fails then though.

Maybe this is only supported in a newer Java version... fixed.

**Review by nit0906 — APPROVED** [review]

**Comment by mreutegg** [comment]
@thomasmueller there is a test failure: https://ci-builds.apache.org/job/Jackrabbit/job/oak-trunk-pr/job/PR-532/2/testReport/junit/org.apache.jackrabbit.oak.plugins.index.lucene.directory/ChunkedOakDirectoryTest/oak_lucene___writes_CustomBlobSize/?cloudbees-analytics-link=scm-reporting%2Ftests%2Ffailed

I've also seen this reproduce on another environment.

**Comment by thomasmueller** [comment]
@mreutegg thanks! I'm sorry I didn't see this... Fixed with https://github.com/apache/jackrabbit-oak/commit/92c408ea4ef1f345a028f60d4624666c34d11013

**Pull request merged**